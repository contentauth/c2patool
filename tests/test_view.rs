mod common;

use common::{cmd, test_img_path};
use insta_cmd::assert_cmd_snapshot;

#[test]
fn test_view_manifest() {
    assert_cmd_snapshot!(cmd().arg("view").arg("manifest").arg(test_img_path()));
}

// TODO: https://github.com/mitsuhiko/insta-cmd/issues/15
#[test]
fn test_view_manifest_debug() {
    // assert_cmd_snapshot!(cmd()
    //     .arg("view")
    //     .arg("manifest")
    //     .arg("--debug")
    //     .arg(test_img_path()));
}

// TODO: https://github.com/mitsuhiko/insta-cmd/issues/15
#[test]
fn test_view_ingredient() {
    // assert_cmd_snapshot!(cmd().arg("view").arg("ingredient").arg(test_img_path()));
}

#[test]
fn test_view_info() {
    assert_cmd_snapshot!(cmd().arg("view").arg("info").arg(test_img_path()));
}

#[test]
fn test_view_manifest_info() {
    assert_cmd_snapshot!(cmd().arg("view").arg("tree").arg(test_img_path()));
}

// TODO: https://github.com/mitsuhiko/insta-cmd/issues/15
#[test]
fn test_view_manifest_certs() {
    // let mut binding = cmd();
    // let mut cmd = binding.arg("view").arg("certs").arg(test_img_path());
    // assert_cmd_snapshot!(cmd);
}
