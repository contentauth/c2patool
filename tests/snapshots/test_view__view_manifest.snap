---
source: tests/test_view.rs
info:
  program: c2patool
  args:
    - view
    - manifest
    - /Users/nsalvemini/Documents/repos/c2patool/tests/fixtures/C.jpg
---
success: true
exit_code: 0
----- stdout -----
{
  "active_manifest": "contentauth:urn:uuid:a621a5d8-147d-499b-95d7-2eaff1009efc",
  "manifests": {
    "contentauth:urn:uuid:a621a5d8-147d-499b-95d7-2eaff1009efc": {
      "claim_generator": "make_test_images/0.6.1 c2pa-rs/0.6.1",
      "title": "C.jpg",
      "format": "image/jpeg",
      "instance_id": "xmp:iid:3de8baae-4de5-4961-8b5a-441bb699bf1b",
      "thumbnail": {
        "format": "image/jpeg",
        "identifier": "self#jumbf=c2pa.assertions/c2pa.thumbnail.claim.jpeg"
      },
      "ingredients": [],
      "assertions": [
        {
          "label": "stds.schema-org.CreativeWork",
          "data": {
            "@context": "http://schema.org/",
            "@type": "CreativeWork",
            "author": [
              {
                "@type": "Person",
                "name": "Gavin Peacock"
              }
            ]
          },
          "kind": "Json"
        },
        {
          "label": "c2pa.actions",
          "data": {
            "actions": [
              {
                "action": "c2pa.created"
              },
              {
                "action": "c2pa.drawing",
                "parameters": {
                  "name": "gradient"
                }
              }
            ]
          }
        }
      ],
      "signature_info": {
        "alg": "Ps256",
        "issuer": "C2PA Test Signing Cert",
        "cert_serial_number": "720724073027128164015125666832722375746636448153"
      },
      "label": "contentauth:urn:uuid:a621a5d8-147d-499b-95d7-2eaff1009efc"
    }
  }
}

----- stderr -----
